consensus_test.go

```diff
@@ -459,28 +459,20 @@ func TestTransition_ValidateState_MoveToCommitState(t *testing.T) {
 		m := newMockPbft(t, []NodeID{"A", "B", "C", "D"}, nil, "A")
 		m.setState(ValidateState)
 
-		// Prepare messages
-		m.emitMsg(createMessage(NodeID("A"), MessageReq_Prepare, nil))
-		m.emitMsg(createMessage(NodeID("B"), MessageReq_Prepare, nil))
-		// repeated message is not included
-		m.emitMsg(createMessage(NodeID("B"), MessageReq_Prepare, nil))
-		m.emitMsg(createMessage(NodeID("C"), MessageReq_Prepare, nil))
-
 		// Commit messages
 		m.emitMsg(createMessage(NodeID("B"), MessageReq_Commit, nil))
 		m.emitMsg(createMessage(NodeID("C"), MessageReq_Commit, nil))
+		m.emitMsg(createMessage(NodeID("D"), MessageReq_Commit, nil))
 
 		m.runCycle(context.Background())
 
 		m.expect(expectResult{
-			sequence:               1,
-			state:                  CommitState,
-			prepareMsgs:            3,
-			commitMsgs:             3, // Commit messages (A proposer sent commit via state machine loop, C and D sent commit via emit message)
-			commitMsgsVotingPower:  3,
-			locked:                 true,
-			outgoing:               1, // A commit message
-			prepareMsgsVotingPower: 3,
```

Comment:
Yes, this is also due to the fact that messages are processed in parallel and we don't (yet) have mechanism to wait for all the emitted messages get processed until we make assertions. 

Therefore most of the time manages to process `commit` messages before the `prepare` messages (since `commit` message type takes precedence over `prepare` if both are for the same sequence and round, because of `iota` value and message queue sorting logic).
So that's why I've opted to emit just a commit messages and bypass the problem.